File: 0074-search-a-2d-matrix/0074-search-a-2d-matrix.cpp
class Solution {
public:
    bool searchMatrix(vector<vector<int>>& matrix, int target) {
        int row=-1;
        for(int i=0;i<matrix.size();i++){
            // cout<<matrix[i][0];
            if(matrix[i][0]<=target && matrix[i][matrix[i].size()-1]>=target)
                row=i;
        }
        if(row!=-1){
            for(int i=0;i<matrix[row].size();i++){
                if(matrix[row][i]==target)
                    return true;
            }
        }
        return false;
    }
};

File: 0074-search-a-2d-matrix/0074-search-a-2d-matrix.java
class Solution {
    public boolean searchMatrix(int[][] matrix, int target) {
        for(int i=0;i<matrix.length;i++){
            if(target<=matrix[i][matrix[0].length-1]){
                int start=0,end=matrix[0].length-1,mid;
                while(start<=end){
                    mid=start+(end-start)/2;
                    if(matrix[i][mid]==target) return true;
                    else if(matrix[i][mid]<target) start=mid+1;
                    else end=mid-1;
                }
                return false;
            }
        }
        return false;
    }
}